# Hypothesis testing(Customer+order form)

## Null hypothesis(H=0):-The defective %  varies by centre.

## Alternative hypothesis(H!=0):-The defective %is not varies by centre.

In [1]:
## Loading laibaries
import pandas as pd
import numpy as np
from scipy import stats
from scipy.stats import norm
from scipy.stats import chi2_contingency

In [2]:
## loading the data set
df=pd.read_csv('Costomer+OrderForm.csv')

In [3]:
df.head()

Phillippines   Indonesia       Malta       India
0   Error Free  Error Free   Defective  Error Free
1   Error Free  Error Free  Error Free   Defective
2   Error Free   Defective   Defective  Error Free
3   Error Free  Error Free  Error Free  Error Free
4   Error Free  Error Free   Defective  Error Free

In [4]:
df.shape

(300, 4)

### extracting no of error free and defective from each column.

In [5]:
## checking for phillipines
df['Phillippines'].value_counts()

Error Free    271
Defective      29
Name: Phillippines, dtype: int64

In [6]:
## checking for Indonesia
df['Indonesia'].value_counts()

Error Free    267
Defective      33
Name: Indonesia, dtype: int64

In [7]:
## checking for Malta
df['Malta'].value_counts()

Error Free    269
Defective      31
Name: Malta, dtype: int64

In [8]:
## checking for India
df['India'].value_counts()

Error Free    280
Defective      20
Name: India, dtype: int64

In [9]:
# Make a contingency table
ot=np.array([[271,267,269,280],[29,33,31,20]])
ot

array([[271, 267, 269, 280],
       [ 29,  33,  31,  20]])

In [10]:
# Chi2 contengency independence test
chi2_contingency(ot) # o/p is (Chi2 stats value, p_value, df, expected obsvations)

(3.858960685820355,
 0.2771020991233135,
 3,
 array([[271.75, 271.75, 271.75, 271.75],
        [ 28.25,  28.25,  28.25,  28.25]]))

## From above result our p-value i.e 0.2771> 0.05 so we accept our null hypothesis.

## Hence,The defective % varies by centre.